samila/functions.py

```diff
@@ -95,17 +95,70 @@ def distance_calc(s1, s2):
     return distances[-1]
 
 
-def filter_color(color):
+def is_valid_color(color):
+    """
+    Check that input color format is valid or not.
+
+    :param color: given color
+    :type color: any format
+    :return: result as bool
+    """
+    try:
+        _ = matplotlib.colors.to_hex(color)
+        return True
+    except ValueError:
+        return False
+
+
+def color_complement(color):
+    """
+    Calculate complement color.
+
+    :param color: given color
+    :type color: any format
+    :return: complement color
+    """
+    color = color[1:]
+    color = int(color, 16)
+    comp_color = 0xFFFFFF ^ color
+    comp_color = "#%06X" % comp_color
+    return comp_color
+
+
+def filter_color(color, bgcolor):
     """
     Filter given color and return it.
 
     :param color: given color
-    :type color: str or tuple
-    :return: filtered version of color
+    :type color: any format
+    :param bgcolor: giver background color
```

Comment:
`given`

Reply:
fixed in https://github.com/sepandhaghighi/samila/pull/131/commits/17d8cf95f14f968890c1ce71d24b1c755b3a99cb